src/storage/v2/inmemory/storage.cpp

```diff
@@ -670,25 +670,19 @@ utils::BasicResult<StorageManipulationError, void> InMemoryStorage::InMemoryAcce
     // Replica can log only the write transaction received from Main
     // so the Wal files are consistent
     if (replState.IsMain() || desired_commit_timestamp.has_value()) {
-      could_replicate_all_sync_replicas = mem_storage->AppendToWalDataDefinition(transaction_, *commit_timestamp_);
-      // Take committed_transactions lock while holding the engine lock to
-      // make sure that committed transactions are sorted by the commit
-      // timestamp in the list.
-      mem_storage->committed_transactions_.WithLock([&](auto & /*committed_transactions*/) {
-        // TODO: release lock, and update all deltas to have a local copy
-        // of the commit timestamp
-        MG_ASSERT(transaction_.commit_timestamp != nullptr, "Invalid database state!");
-        transaction_.commit_timestamp->store(*commit_timestamp_, std::memory_order_release);
-        // Replica can only update the last commit timestamp with
-        // the commits received from main.
-        if (replState.IsMain() || desired_commit_timestamp.has_value()) {
-          // Update the last commit timestamp
-          mem_storage->repl_storage_state_.last_commit_timestamp_.store(*commit_timestamp_);
-        }
-        // Release engine lock because we don't have to hold it anymore
-        // and emplace back could take a long time.
-        engine_guard.unlock();
-      });
+      could_replicate_all_sync_replicas =
+          mem_storage->AppendToWalDataDefinition(transaction_, *commit_timestamp_);  // protected by engine_guard
+      // TODO: release lock, and update all deltas to have a local copy of the commit timestamp
+      MG_ASSERT(transaction_.commit_timestamp != nullptr, "Invalid database state!");
+      transaction_.commit_timestamp->store(*commit_timestamp_, std::memory_order_release);  // protected by engine_guard
+      // Replica can only update the last commit timestamp with
+      // the commits received from main.
+      if (replState.IsMain() || desired_commit_timestamp.has_value()) {
```

Comment:
Same check as in the outside if scope (line 672)

Reply:
Outside scope of this PR, maybe change as part of replication work